autoload/sy/repo.vim

```diff
@@ -73,13 +73,23 @@ function! sy#repo#get_diff_start(vcs) abort
     endif
 
     let [cmd, options] = s:initialize_job(a:vcs)
+    let [cwd, chdir] = sy#util#chdir()
 
     call sy#verbose(printf('CMD: %s | CWD: %s', string(cmd), b:sy.info.dir), a:vcs)
+
+    try
+      execute chdir fnameescape(b:sy.info.dir)
+    catch
+      echohl ErrorMsg
+      echomsg 'signify: Switching directory failed: '. b:sy.info.dir
+      echohl NONE
+      return
+    endtry
     let b:sy_job_id_{a:vcs} = jobstart(cmd, extend(options, {
-          \ 'cwd':       b:sy.info.dir,
           \ 'on_stdout': function('s:callback_nvim_stdout'),
           \ 'on_exit':   function('s:callback_nvim_exit'),
           \ }))
+    execute chdir '-'
```

Comment:
Here we can use `:cd -` because that even works if the prior `cd ...` did change to the current directory again.

E.g.

```vim
:cd ~
:cd /tmp
:cd /tmp
:cd -
" still in /tmp
```

Reply:
`-` isn't needed, though.  We already have the current directory in the return from `sy#util#chdir()`.

Reply:
Hmm, I somehow thought that `cd -` would stress the fact that we only changed the directory temporarily and thus make it more readable. (But `sy#util#chdir()` should be changed then not to return the cwd indeed.)